distributedlog-protocol/src/main/java/com/twitter/distributedlog/EnvelopedRecordSetWriter.java

```diff
@@ -139,42 +145,43 @@ public synchronized ByteBuffer getBuffer() {
     }
 
     ByteBuffer createBuffer() {
-        byte[] data = buffer.getData();
+        ByteBuffer data = buffer.getData();
         int dataOffset = HEADER_LEN;
         int dataLen = buffer.size() - HEADER_LEN;
 
         if (COMPRESSION_CODEC_LZ4 != codecCode) {
-            ByteBuffer recordSetBuffer = ByteBuffer.wrap(data, 0, buffer.size());
+            ByteBuffer recordSetBuffer = data;
```

Comment:
i think it would be cleaner if you duplicate() the buffer.. its a little confusing to simply create a new reference.